contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/HadoopStoreBuilder.java

```diff
@@ -409,6 +380,69 @@ public void build() {
     }
 
     /**
+     * This function configures a {@link JobConf} instance based on the {@param size} of the data
+     * that Voldemort will need to serve. If the size provided is that of the input data, then
+     * these settings will be calculated in a best-effort basis. It is not possible to predict in
+     * advance the correct settings based on the input data, since compression factors may vary
+     * a lot between data sets.
+     *
+     * On the other hand, if the size is that of the mappers' output, then it should be possible
+     * to predict proper settings. There is still another edge case, however, which we cannot
+     * avoid completely: if the data is heavily skewed towards a hot partition, then it may not
+     * be possible at all to tweak these settings correctly.
+     *
+     * @param size of the dataset
+     * @param conf MapReduce {@link JobConf} for which to set properties.
+     * @return
+     */
+    private void setNumChunksAndNumReduceTasks(long size, JobConf conf) {
+        logger.info("Data size = " + size +
+                ", replication factor = " + storeDef.getReplicationFactor() +
+                ", numNodes = " + cluster.getNumberOfNodes() +
+                ", numPartitions = " + cluster.getNumberOfPartitions() +
+                ", chunk size = " + chunkSizeBytes);
+
+        // Base numbers of chunks and reducers, will get modified according to various settings
+        int numChunks = (int) (size / cluster.getNumberOfPartitions() / chunkSizeBytes) + 1; /* +1 so we round up */
```

Comment:
Note that the +1 is added on this line.  The refactoring of this method makes it harder to see that it is a change.